### PR TITLE
patch (DOCS-CODE-016): [authN] [sign-in] ensure Azure AD v2.0 usage in the canonical webapp

### PR DESCRIPTION
**Pull request by ferantivero**
related: #508094, #1652433

## WHY

we wanted to be sure our users understand the difference between scaffolding a web app using Microsoft Identity Web vs the legacy one using `Microsoft.AspNetCore.Authentication.AzureAD.UI` which is used to create an Azure AD v1.0 application  

## WHAT Changed?

- prescribe how to create an Microsoft identity platform (Azure AD v2.0) application.
- streamline how to add authN.
- modify code to be aligned with other `Azure AD v2.0` examples out there.

## HOW to Test?

follow the `README.md` file under the sing-in scenario.

### CODE REVIEW

File: src/sign-in-webapp/README.md [hunk 1]
```diff
@@ -69,12 +69,20 @@ Later the web app can make calls to the Microsoft 365 Graph API on behalf of sig
 
 ## Scaffold the web app by using the ASP.NET Core Razor project template
 
-1. Execute the following command to create the new web api project
+1. Install the latest `Microsoft.Identity.Web` templates from NuGet.org
 
    ```bash
-   dotnet new webapp --auth SingleOrg --aad-instance "https://login.microsoftonline.com/" --client-id ${AZURE_AD_APP_CLIENT_ID_WEBAPP} --tenant-id $(az account show --query tenantId --output tsv) --domain ${AZURE_AD_APP_DOMAIN} --called-api-url "https://graph.microsoft.com/v1.0/me"
+   dotnet new --install Microsoft.Identity.Web.ProjectTemplates::1.21.0
```
Comment (mmacy) [inline src/sign-in-webapp/README.md:75]:
```suggestion
   dotnet new --install Microsoft.Identity.Web.ProjectTemplates
```

But only if you _really_ meant latest. :)

File: src/sign-in-webapp/README.md [hunk 2]
```diff
@@ -69,12 +69,20 @@ Later the web app can make calls to the Microsoft 365 Graph API on behalf of sig
...
    ```
 
+1. Create a Microsoft identity platform (Azure AD v2.0) application. Execute the following command to create the web app project
```
Comment (mmacy) [inline src/sign-in-webapp/README.md:78]:
```suggestion
1. Create a new web application. Execute the following command to create the web app project:
```

File: src/sign-in-webapp/README.md [hunk 3]
```diff
@@ -69,12 +69,20 @@ Later the web app can make calls to the Microsoft 365 Graph API on behalf of sig
...
+
+   ```bash
+   dotnet new webapp2 --auth SingleOrg --aad-instance "https://login.microsoftonline.com/" --client-id ${AZURE_AD_APP_CLIENT_ID_WEBAPP} --tenant-id $(az account show --query tenantId --output tsv) --domain ${AZURE_AD_APP_DOMAIN} --called-api-url "https://graph.microsoft.com/v1.0/me"
```
Comment (mmacy) [inline src/sign-in-webapp/README.md:81]: We should include instructions for only for running the sample application, not creating it.

The tutorials on Docs will walk the customer through adding authentication to a base web app that was generated without using `--auth`.

File: src/sign-in-webapp/README.md [hunk 4]
```diff
@@ -69,12 +69,20 @@ Later the web app can make calls to the Microsoft 365 Graph API on behalf of sig
...
+   ```
+
+   :warning: Today, with ASP.NET Core 3.1 without Microsoft Identity Web, when running `dotnet new --auth` and creating a `webapp` from an ASP.NET core template, the application is targeting the Azure AD v1.0 endpoint. Microsoft Identity Web provides the glue between the ASP.NET Core middleware and MSAL .NET to bring a clearer, more robust developer experience, which also leverages the power of the Microsoft identity platform (formerly Azure AD v2.0 endpoint). `Microsoft.Identity.Web` is a simpler way to use Azure AD in ASP.NET Core web apps and web APIs. It doesn't replace ASP.NET Identity in any way, it doesn't replace AddOpenIdConnect, AddJwtBearer or AddCookie or any of the lower level primitives, but it does use and configure them correctly for Azure AD. It replaces AzureAD.UI and AzureADB2C.UI which are obsolete in .NET 5.0 or later. For more information, please take a look at https://github.com/AzureAD/microsoft-identity-web/wiki/Microsoft-Identity-Web-basics#why-use-microsoftidentityweb
```
Comment (mmacy) [inline src/sign-in-webapp/README.md:84]: You can cut all this.

```suggestion
```

File: src/sign-in-webapp/README.md [hunk 5]
```diff
@@ -69,12 +69,20 @@ Later the web app can make calls to the Microsoft 365 Graph API on behalf of sig
 
 ## Scaffold the web app by using the ASP.NET Core Razor project template
```
Comment (mmacy) [inline src/sign-in-webapp/README.md:70]: New template coming in a few minutes. In the README, they'll learn how to prepare for and set up the sample app (installing packages if they need to) but we won't have them scaffold a _new_ application in the README.

**Review by mmacy — CHANGES_REQUESTED** [review]
New README coming that should help clear some things up - sorry for the delay, @ferantivero!

File: src/sign-in-webapp/README.md (hunk 1 above)
Comment (ferantivero) [reply src/sign-in-webapp/README.md:75]: thanks for the feedback @mmacy, we were including that to create the app but there is an explicit request in here to remove that part.

Therefore, we got rid of it from [3959b](https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/9/commits/3959bbe9d73c26a2f8db3a7f8777d11575ae3d69) via https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/6#discussion_r761570742

File: src/sign-in-webapp/README.md (hunk 3 above)
Comment (ferantivero) [reply src/sign-in-webapp/README.md:81]: thanks for the feedback @mmacy, that ^ sounds reasonable. 

Done  | Addressed from [3959b](https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/9/commits/3959bbe9d73c26a2f8db3a7f8777d11575ae3d69)

File: src/sign-in-webapp/README.md (hunk 5 above)
Comment (ferantivero) [reply src/sign-in-webapp/README.md:70]: thanks for creating that template @mmacy, we're addressing all readme-related changes from another [pr](#9).

**Comment by ferantivero** [comment]
@mmacy merging as-is for now so we can publish other PRs in the queue. 

Please consider following up my replies from here or directly from #9 where all your feedback should've been addressed.

**Comment by ckittel** [comment]
Thanks @ferantivero -- all of @mmacy's comments were addressed, just don't want this one lingering out there much longer.  @mmacy, if you do want to see additional README changes, please work with the new content development flow to make any tweaks as necessary!

File: src/sign-in-webapp/README.md (hunk 4 above)
Comment (ckittel) [reply src/sign-in-webapp/README.md:84]: Just marking this one resolved to remove the PR block.  @mmacy if there is any more feedback beyond the change that @ferantivero made a week ago, please let us know.

File: src/sign-in-webapp/README.md (hunk 3 above)
Comment (ckittel) [reply src/sign-in-webapp/README.md:81]: Just marking this one resolved to remove the PR block.  @mmacy if there is any more feedback beyond the change that @ferantivero made a week ago, please let us know.

File: src/sign-in-webapp/README.md (hunk 2 above)
Comment (ckittel) [reply src/sign-in-webapp/README.md:78]: Just marking this one resolved to remove the PR block.  @mmacy if there is any more feedback beyond the change that @ferantivero made a week ago, please let us know.

File: src/sign-in-webapp/README.md (hunk 1 above)
Comment (ckittel) [reply src/sign-in-webapp/README.md:75]: Just marking this one resolved to remove the PR block.  @mmacy if there is any more feedback beyond the change that @ferantivero made a week ago, please let us know.

**Pull request merged**